### DISCUSSION TITLE
getInvoices always returns NULL

### DISCUSSION THREAD

**Discussion by johnleyburn** [XeroAPI/xero-php-oauth2#323] [Q&A]
I'm fairly new to PHP and Xero but, so far have managed to get everything I need working.  I do, however, have a problem with getInvoices that I just cannot solve.

My code is as follows:

      $config = XeroAPI\XeroPHP\Configuration::getDefaultConfiguration()->setAccessToken($setting['AccToken']);

      $apiInstance = new XeroAPI\XeroPHP\Api\AccountingApi(new GuzzleHttp\Client(), $config);
      $xeroTenantId = $setting['XTenantID'];
      $ifModifiedSince = null;
      $where = null;
      $order = null;
      $iDs = null;
      $invoiceNumbers = null;
      $contactIDs = array($setting["XMCDContactID"]);
      $statuses = null;
      $page = 1;
      $includeArchived = null;
      $createdByMyApp = null;
      $unitdp = null;
      $summaryOnly = true;

      try
      {
        $result = $apiInstance->getInvoices($xeroTenantId, $ifModifiedSince, $where, $order, $iDs, $invoiceNumbers, $contactIDs, $statuses, $page, $includeArchived, $createdByMyApp, $unitdp, $summaryOnly);
      }
      catch (Exception $e)
      {
        echo 'Exception when calling AccountingApi->getInvoices: ', $e->getMessage(), PHP_EOL;
      }

Which, I assume, should return all invoices for the ContactID ($setting["XMCDContactID"] is definitely populated with the correct ID) but it returns NULL.

I've also tried changing the following (both the invoices definitely exist in Xero).

      $invoiceNumbers = array("INV-0044", "INV-0045");
      $contactIDs = null;

But that also returns NULL.

I'm clearly missing something obvious but I just can't see it...

John

**Comment by pumpkinball** [comment]
Hi @johnleyburn you're right your code does look fine.
It may be worth popping in your parameters into [API Explorer](https://api-explorer.xero.com/) and running the same GET request, just to check that it returns what you're expecting there. https://api-explorer.xero.com/accounting/invoices/getinvoices

**Reply by johnleyburn** [reply]
Thanks again for that, I think sometimes you can't see the wood for the trees.  I spent hours looking at this yesterday and, having followed your suggestion and gone back to my code, I suddenly realised that the function that my code was in had no return hence the null!

Sorry for wasting your time, must try harder next time...

John

**Reply by pumpkinball** [reply]
It happens to all of us 😄

**Comment by johnleyburn** [comment]
Thanks for the suggestion, I've just tried the API Explorer and it is returns exactly what I would expect...
